edhoc/roles/responder.py

```diff
@@ -236,6 +236,23 @@ def finalize(self, message_three: bytes) -> Union[Tuple[bytes, bytes, int, int],
 
         return self.msg_1.conn_idi, self._conn_id, app_aead.identifier, app_hash.identifier
 
+    def _verify_signature_or_mac3(self, signature_or_mac3: bytes) -> bool:
+        mac_3 = self._mac(self.cred_idi, self.remote_cred, self._hkdf3, 'K_3m', 16, 'IV_3m', 13, self._th3_input, self._prk4x3m, self.aad3_cb)
+
+        if not self.is_static_dh(self.remote_role):
+            external_aad = self._external_aad(self.remote_cred, self._th3_input, self.aad3_cb)
+            cose_sign = Sign1Message(
+                phdr=self.cred_idi,
+                uhdr={headers.Algorithm: self.cipher_suite.sign_alg},
+                payload=mac_3,
+                external_aad=external_aad)
+            # FIXME peeking into internals (probably best resolved at pycose level)
```

Comment:
Yes I agree, using the `pycose `library here in `edhoc `has uncovered some design flaws in the API and its usage.